File: mo-system/mo-system-service/src/main/java/ink/ckx/mo/system/mapper/SysMenuMapper.java
package ink.ckx.mo.system.mapper;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import ink.ckx.mo.system.api.model.bo.RouteBO;
import ink.ckx.mo.system.api.model.bo.RolePermBO;
import ink.ckx.mo.system.api.model.entity.SysMenu;
import org.apache.ibatis.annotations.Mapper;

import java.util.List;

/**
 * @author chenkaixin
 */
@Mapper
public interface SysMenuMapper extends BaseMapper<SysMenu> {

    /**
     * 获取路由列表
     *
     * @return
     */
    List<RouteBO> listRoutes();

    /**
     * 权限列表
     *
     * @return
     */
    List<RolePermBO> listPermission();
}

File: mo-admin/mo-admin-service/src/main/kotlin/ink/ckx/mo/admin/mapper/SysMenuMapper.kt
package ink.ckx.mo.admin.mapper

import com.baomidou.mybatisplus.core.mapper.BaseMapper
import ink.ckx.mo.admin.api.model.bo.RolePermBO
import ink.ckx.mo.admin.api.model.bo.RouteBO
import ink.ckx.mo.admin.api.model.entity.SysMenu
import org.apache.ibatis.annotations.Mapper

/**
 * @author chenkaixin
 */
@Mapper
interface SysMenuMapper : BaseMapper<SysMenu> {

    /**
     * 获取路由列表
     *
     * @return
     */
    fun listRoutes(): List<RouteBO>

    /**
     * 权限列表
     *
     * @return
     */
    fun listPermission(): List<RolePermBO>
}